src/NetAnalyzers/Directory.Build.props

```diff
@@ -0,0 +1,10 @@
+<!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
+<Project>
+  <Import Project="..\Directory.Build.props"/>
+
+  <PropertyGroup>
+    <FxCopAnalyzersProject Condition="$(MSBuildProjectName.StartsWith('Microsoft.CodeQuality')) OR $(MSBuildProjectName.StartsWith('Microsoft.NetCore')) OR $(MSBuildProjectName.StartsWith('Microsoft.NetFramework'))">true</FxCopAnalyzersProject>
+	<DefineConstants Condition="'$(FxCopAnalyzersProject)' == 'true'">$(DefineConstants),FXCOP_ANALYZERS</DefineConstants>
```

Comment:
What's the purpose of these constants?

Reply:
`FXCOP_ANALYZERS` - used for our currently shipping FxCopAnalyzers package, where we want to retain current packaging names, rule severities and enabled by default values.

`NET_ANALYZERS` - used for new NetAnalyzers package, where we want new values as per our triage.

Both these packages share the same underlying source files, so we need some preprocessor directives to distinguish and generate respective severities and enabled by default values.

Reply:
I see. So basically this is just to single out MS.CA.*.NetAnalyzers